### PR TITLE
BE | Ask va api: update inquiry payload mapping

### PR DESCRIPTION
**Pull request by Khoa-V-Nguyen**
## Summary

This PR introduces several updates and fixes to improve functionality, performance, and maintainability across the codebase:
1. Enhancements to Inquiry Functionality:
	• Updated InquiriesController#create_reply to support attachments.
	• Improved Inquiries::Retriever to include ICN/ID keys in the inquiry payload.
	• Made minor fixes to InquiryPayload and SubmitterProfile.
2. Translation and Structure Updates:
	• Updated the dictionary for translations.
	• Fixed the structure in the en.yml file for better consistency.
3. Attachment Support:
	• Enhanced Correspondences::Creator to include support for attachments.
4. Performance Optimization:
	• Memoized profile in va_0873.rb to improve performance.

These changes collectively enhance the functionality of inquiries and correspondence handling while optimizing performance and maintaining clarity in code structure.

## Related issue(s)

- https://github.com/department-of-veterans-affairs/ask-va/issues/1465

## Testing done

- [x] *New code is covered by unit tests*

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [x]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x]  Feature/bug has a monitor built into Datadog (if applicable)
- [x]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected

### CODE REVIEW

**Review by hemeshvpatel — APPROVED** [review]
lgtm

**Comment by jla3378** [comment]
Making sure this doesn't get lost: https://github.com/department-of-veterans-affairs/vets-api/issues/19663

**Review by hemeshvpatel — APPROVED** [review]
lgtm

**Review by joehall-tw — APPROVED** [review]
Approving after small change.

File: app/models/form_profiles/va_0873.rb [hunk 1]
```diff
@@ -64,24 +63,40 @@ def initialize_ava_profile
 
     VA0873::FormAvaProfile.new(payload)
   rescue => e
-    log_exception_to_sentry(e, {}, prefill: :personal_information)
-    {}
+    handle_exception(e, :ava_profile)
```
Comment (rjohnson2011) [inline app/models/form_profiles/va_0873.rb:66]: Is there any risk of PII being exposed in this Sentry log?

File: app/models/form_profiles/va_0873.rb [hunk 2]
```diff
@@ -64,24 +63,40 @@ def initialize_ava_profile
...
+    handle_exception(e, :preferred_name)
   end
 
+  # Retrieves the profile from the AskVAApi service
   def profile
-    AskVAApi::Profile::Retriever.new(icn: user.icn).call
+    @profile ||= AskVAApi::Profile::Retriever.new(icn: user.icn).call
   rescue => e
-    log_exception_to_sentry(e, {}, prefill: :personal_information)
+    handle_exception(e, :profile)
```
Comment (rjohnson2011) [inline app/models/form_profiles/va_0873.rb:80]: Any potential for PII being logged to Sentry?

File: app/models/form_profiles/va_0873.rb [hunk 3]
```diff
@@ -36,24 +36,23 @@ def prefill
...
+    service_number = profile.is_a?(Hash) ? profile[:service_number] : profile.service_number
 
+    # Merge the user's normalized name, preferred name, and service number into the payload
     payload = user.full_name_normalized.merge(preferred_name:, service_number:)
 
     VA0873::FormPersonalInformation.new(payload)
   rescue => e
-    log_exception_to_sentry(e, {}, prefill: :personal_information)
-    {}
+    handle_exception(e, :personal_information)
```
Comment (rjohnson2011) [inline app/models/form_profiles/va_0873.rb:50]: Same question about PII

File: app/models/form_profiles/va_0873.rb (hunk 2 above)
Comment (Khoa-V-Nguyen) [reply app/models/form_profiles/va_0873.rb:80]: None, it'll just log the error. No PII will be in the error message.

File: app/models/form_profiles/va_0873.rb (hunk 3 above)
Comment (Khoa-V-Nguyen) [reply app/models/form_profiles/va_0873.rb:50]: None, it'll just log the error. No PII will be in the error message.

File: app/models/form_profiles/va_0873.rb (hunk 1 above)
Comment (Khoa-V-Nguyen) [reply app/models/form_profiles/va_0873.rb:66]: None, it'll just log the error. No PII will be in the error message.

**Review by rjohnson2011 — APPROVED** [review]

**Pull request merged**